File: CPP/Program/Spin.cpp
#ifndef SPIN_CPP
#define SPIN_CPP
#include <iostream>
#include <string>

#include "BasicMove.cpp"

using namespace std;

class Spin : public BasicMove{
    private:
        int spinSpeed; //Rotation per minute
    public:
        Spin(int speed, string name, int duration, string difficulty) : BasicMove(name, duration, difficulty){}

        void setSpinSpeed(int speed){
            this->spinSpeed = speed;
        }
        int getSpinSpeed(){
            return this->spinSpeed;
        }

        ~Spin(){}
};

#endif

File: Python/Program/Spin.py
from BasicMove import BasicMove

class Spin(BasicMove):
    def __init__(self, name, duration, difficulty, speed):
        super().__init__(name, duration, difficulty)
        self.__spinSpeed = speed
    
    def setSpinSpeed(self, speed):
        self.__spinSpeed = speed
    def getSpinSpeed(self):
        return self.__spinSpeed